mountpoint-s3/src/cli.rs

```diff
@@ -511,9 +519,46 @@ impl CliArgs {
         filesystem_config.cache_config = self.cache_config();
         filesystem_config.mem_limit = self.mem_limit();
         filesystem_config.use_upload_checksums = self.should_use_upload_checksum(s3_personality);
+
+        if let Some(username) = &self.run_as_user {
+            match Self::resolve_user_group(username) {
+                Ok((uid, gid)) => {
+                    filesystem_config.uid = uid; // Sets filesystem operations to run as this user
+                    filesystem_config.gid = gid;
+                    tracing::info!("Running Mountpoint as user: {}", username);
+
+                    // Convert to proper types for setuid/setgid
+                    let gid = Gid::from_raw(gid);
+                    let uid = Uid::from_raw(uid);
+
+                    // Handle errors
+                    if let Err(e) = setgid(gid) {
+                        tracing::error!(error=?e, "Failed to set GID");
+                    }
+                    if let Err(e) = setuid(uid) {
+                        tracing::error!(error=?e, "Failed to set UID");
+                    }
+                }
+                Err(e) => {
+                    tracing::error!(error=?e, username=%username, "Failed to resolve user/group");
+                }
+            }
+        }
+
         filesystem_config
     }
 
+    fn resolve_user_group(username: &str) -> anyhow::Result<(u32, u32)> {
+        // Get user info using the safe nix API
+        let user = nix::unistd::User::from_name(username)?.ok_or_else(|| anyhow!("User '{}' not found", username))?;
```

Comment:
Nit: Could we import `User` here and just do `User::from_name` (and the same for Group below?)